tpchgen-arrow/src/lineitem.rs

```diff
@@ -0,0 +1,182 @@
+///   "| 1          | 15635     | 638       | 6            | 32.00      | 49620.16        | 0.07       | 0.02  | N            | O            | 1996-01-30 | 1996-02-07   | 1996-02-03    | DELIVER IN PERSON | MAIL       | arefully slyly ex                   |",
+///   "| 2          | 106170    | 1191      | 1            | 38.00      | 44694.46        | 0.00       | 0.05  | N            | O            | 1997-01-28 | 1997-01-14   | 1997-02-02    | TAKE BACK RETURN  | RAIL       | ven requests. deposits breach a     |",
+///   "| 3          | 4297      | 1798      | 1            | 45.00      | 54058.05        | 0.06       | 0.00  | R            | F            | 1994-02-02 | 1994-01-04   | 1994-02-23    | NONE              | AIR        | ongside of the furiously brave acco |",
+///   "| 3          | 19036     | 6540      | 2            | 49.00      | 46796.47        | 0.10       | 0.00  | R            | F            | 1993-11-09 | 1993-12-20   | 1993-11-24    | TAKE BACK RETURN  | RAIL       |  unusual accounts. eve              |",
+///   "| 3          | 128449    | 3474      | 3            | 27.00      | 39890.88        | 0.06       | 0.07  | A            | F            | 1994-01-16 | 1993-11-22   | 1994-01-23    | DELIVER IN PERSON | SHIP       | nal foxes wake.                     |",
+///   "+------------+-----------+-----------+--------------+------------+-----------------+------------+-------+--------------+--------------+------------+--------------+---------------+-------------------+------------+-------------------------------------+"
+/// ]);
+/// ```
+// # TODOs:
+// 1. create individual column iterators to avoid a copy into rows
+// 2. Maybe Recycle buffers (don't reallocate new ones all the time) :thinking:
+// Based off code / types from DataFusion
+// https://github.com/apache/datafusion/blob/a1ae15826245097e7c12d4f0ed3425b25af6c431/benchmarks/src/tpch/mod.rs#L104-L103
+pub struct LineItemArrow {
+    inner: LineItemGeneratorIterator<'static>,
+    batch_size: usize,
+}
+
+impl LineItemArrow {
+    pub fn new(generator: LineItemGenerator<'static>) -> Self {
+        Self {
+            inner: generator.iter(),
+            batch_size: DEFAULT_BATCH_SIZE,
+        }
+    }
+
+    /// Set the batch size
+    pub fn with_batch_size(mut self, batch_size: usize) -> Self {
+        self.batch_size = batch_size;
+        self
+    }
+}
+
+impl RecordBatchIterator for LineItemArrow {
+    fn schema(&self) -> &SchemaRef {
+        &LINEITEM_SCHEMA
+    }
+}
+
+impl Iterator for LineItemArrow {
+    type Item = RecordBatch;
+
+    /// Generate the next batch of data, if there is one
+    fn next(&mut self) -> Option<Self::Item> {
+        // Get some rows to convert
+        let rows: Vec<_> = self.inner.by_ref().take(self.batch_size).collect();
+        if rows.is_empty() {
+            return None;
+        }
+
+        // Convert column by column
```

Comment:
I a quite pleased at how this works -- it basically follows @scsmithr 's model of buffering `LineItem` rows and then converting column by column

I suspect we can make it even faster if we made per-column data generators instead and skipped the row generation, but I'll maybe look into that as a follow on PR